backend/src/user/user.service.ts

```diff
@@ -171,4 +219,89 @@ export class UserService {
     });
     return res;
   }
+
+  /**
+   * Sends the user a reset password email and returns the JWT token
+   * @param email the email account that requested for a password reset
+   */
+
+  async resetPassword(email: string) {
+    // find user via email provided
+    const [err, user] = await this.findByEmail(email);
+    const username = user.username;
+
+    // if user doesn't exist, do nothing. Just bring user to email sent page
+    if (err || !user) {
+      return;
+    }
+
+    // check if email in cache, print out that the reset mail has already been sent
+    const isEmailInCache = !!(await this.cache.get(
+      RESET_PASSWORD_EMAIL_PREFIX + email
+    ));
+    if (isEmailInCache) {
+      throw new ForbiddenException(AUTH_ERROR.RESET_EMAIL_ALREADY_SENT);
+    }
+
+    //Reset password
+    const resetPasswordVerificationToken = RESET_PASSWORD_PREFIX + v4();
+    await this.cache.set<CacheableResetEmail>(resetPasswordVerificationToken, {
+      email,
+    });
+
+    //Generate a random temporary password for user of length 16
+    const temporaryPassword = randomUUID().slice(0,16);
+
+    const [error, userToReset] = await this.updateResetPassword(
+      user.id,
+      temporaryPassword
+    );
```

Comment:
error returned should be handled in case error occurred while trying to update user's hash aka password